nucliadb_vectors/src/data_point/mod.rs

```diff
@@ -53,6 +56,21 @@ impl DeleteLog for NoDLog {
     }
 }
 
+#[derive(Default, Debug, Clone, Copy, Serialize, Deserialize)]
+pub enum Similarity {
+    Dot,
+    #[default]
+    Cosine,
+}
+impl Similarity {
+    pub fn compute(&self, x: &[u8], y: &[u8]) -> f32 {
+        match self {
```

Comment:
DOUBT: why are you using an enum and branching over all possible variations instead of having a trait and use concrete implementation for each similarity?

Reply:
Is dynamic dispatch slower than a `match` statement for all possible `Similarity` variants?

Reply:
Very much so. Dynamic dispatch requires for the compiler to include a table during runtime that figures out which method are you calling at each given time, and optimising that is actually harder for a compiler. On the other hand, a match expression can be very easy to optimised and actually is very cheap. 
Trait objects are very useful for adding ergonomics or writing code where the types that need to be handle are not a close set (because of user define types for example), but for code where the variants are clear I would say is almost always a good idea to use enums.

Reply:
Mm... That's interesting!